src/components/screens/sprint-view-screen.tsx

```diff
@@ -1,13 +1,242 @@
+import { useState, useEffect } from "react";
+import { Card, CircularProgress, Typography, Box, FormControlLabel, Switch } from "@mui/material";
+import { useLambdasApi } from "src/hooks/use-api";
+import type { Person } from "src/generated/client";
+import { useAtomValue, useSetAtom } from "jotai";
+import { personsAtom } from "src/atoms/person";
+import config from "src/app/config";
+import { userProfileAtom } from "src/atoms/auth";
+import type {
+  Allocations,
+  Projects,
+  TimeEntries
+} from "src/generated/homeLambdasClient/models/";
+import { DataGrid } from "@mui/x-data-grid";
+import { getHoursAndMinutes, getSprintEnd, getSprintStart } from "src/utils/time-utils";
+import TaskTable from "src/components/sprint-view-table/tasks-table";
 import strings from "src/localization/strings";
+import sprintViewProjectsColumns from "src/components/sprint-view-table/sprint-projects-columns";
+import { errorAtom } from "src/atoms/error";
+import {
+  calculateWorkingLoad,
+  totalAllocations,
+  filterAllocationsAndProjects
+} from "src/utils/sprint-utils";
+import { TaskStatusFilter } from "src/components/sprint-view-table/menu-Item-filter-table";
 
 /**
- * Sprint View screen component
+ * Sprint view screen component
  */
 const SprintViewScreen = () => {
-  
+  const { allocationsApi, projectsApi, timeEntriesApi } = useLambdasApi();
+  const persons: Person[] = useAtomValue(personsAtom);
+  const userProfile = useAtomValue(userProfileAtom);
+  const loggedInPerson = persons.find(
+    (person: Person) =>
+      person.id === config.person.forecastUserIdOverride || person.keycloakId === userProfile?.id
+  );
+  const [allocations, setAllocations] = useState<Allocations[]>([]);
+  const [projects, setProjects] = useState<Projects[]>([]);
+  const [timeEntries, setTimeEntries] = useState<number[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [myTasks, setMyTasks] = useState(true);
+  const [filter, setFilter] = useState("");
+  const todaysDate = new Date().toISOString();
+  const sprintStartDate = getSprintStart(todaysDate);
+  const sprintEndDate = getSprintEnd(todaysDate);
+  const columns = sprintViewProjectsColumns({ allocations, timeEntries, projects });
+  const setError = useSetAtom(errorAtom);
+
+  /**
+   * Get project data if user is logged in
+   */
+  useEffect(() => {
+    fetchProjectDetails();
+  }, [loggedInPerson]);
+
+  /**
+   * Fetch allocations, project names and time entries
+   */
+  const fetchProjectDetails = async () => {
+    setLoading(true);
+    if (!loggedInPerson) return;
```

Comment:
Perhaps the loading should be set after the condition otherwise it looks like a potentially stuck loading state